### PR TITLE
update recipe for version 3.3.1

### PR DESCRIPTION
**Pull request by newville**
this updates the recipe to pull version 3.3.1 from github.com

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  fn: {{ name }}-{{ version }}.tar.gz
-  url: https://github.com/pyepics/{{ name }}/archive/{{ version }}.tar.gz
-  sha256: {{ sha256 }}
+  git_rev: {{ version }}
+  git_url: https://github.com/pyepics/pyepics.git
```
Comment (CJ-Wright) [inline recipe/meta.yaml]: Please pull from tarballs not git https://conda-forge.org/docs/meta.html#build-from-tarballs-not-repos

**Comment by CJ-Wright** [comment]
I think you need to have `setuptools` in the `run` section if the code is using `pkg_resources`

**Comment by CJ-Wright** [comment]
@newville when this passes please ping me, I'll merge. Also feel free to add yourself as a maintainer. (I would do that after travis passes and with `[ci skip]` in the first line of the commit, that way we don't wait for all the CI services again)

**Comment by newville** [comment]
@CJ-Wright No idea why the Travis CI is hung -- seems like a bogus problem to me.  

Is it important to have Travis CI run 'conda build' on OSX?  It looks like travis CI is spinning up OSX, installing miniconda and conda-forge-build-setup all to run "conda build recipe".  Is that right? 

Meanwhile, I put pyepics 3.3.1 for Py27, 35, and 36 on a different channel (GSECARS) without incident.  So, at the moment, I'm not really loving conda-forge. I don't really care if I am listed as maintainer of this recipe.  But the current version of pyepics on conda-forge is very out of date and should either be updated or retired.

**Comment by CJ-Wright** [comment]
@conda-forge-admin, please rerender
(This is needed since this PR has removed the windows skip, the rerender will create a build setup for appveyor, which is what conda-forge uses to build windows packages).

**Comment by newville** [comment]
Sorry I have no idea what the rerender thing is or how that works....

**Comment by CJ-Wright** [comment]
Do you mind if I push to your branch?

**Comment by newville** [comment]
@CJ-Wright  Go right ahead... You can totally take this over!

**Comment by isuruf** [comment]
@conda-forge-admin, please rerender

**Comment by CJ-Wright** [comment]
@newville fyi here is a feed of Travis' status, they are running in reduced macOS capacity. https://www.traviscistatus.com/

**Comment by newville** [comment]
@CJ-Wright OK, thanks for the update.  I don't have a strong opinion on how to proceed.  The problem seems to be with which services conda-forge chooses to build packages.   I can build (have built) conda packages for multiple versions and platforms. The approach used seems unnecessarily complicated for a pure-python package to me, but maybe I'm not seeing all facets of the problem.   Perhaps the simplest strategy is just to wait until the services are back online....

As it happens, I'll be visiting NSLS-II much of the week, and may not have time to work on this.

**Comment by CJ-Wright** [comment]
I think the main complication is that it ships with it's own pre-compiled C libraries and is why `noarch` failed, I think. I'll be at the NSLS-II myself this week if you want to talk in person.

**Comment by mbargull** [comment]
@CJ-Wright: you are right, the package cannot be `noarch: python` as it is, due to the C libs, not an architecture-independent Python package. It only (unnecessarily) bundles compiled libraries for platforms other than the one it's getting installed on -- that doesn't make it platform independent (and, for different platforms, e.g., `aarch64`, it would miss those libraries).
@newville: Not directly related to this `conda-forge` recipe, but I'd suggest to change https://github.com/pyepics/pyepics/blob/3.3.1/setup.py#L47-L48 to conditionally include only the libraries for the current platform when creating binary distributions or during installation. You can do that, for example, by using custom classes `EpicsInstall`/`EpicsBDist` derived from `setuptools.command.install.install`/`distutils.command.bdist.bdist` that pick only the current platform-specific libraries and then use them via `setup(..., cmdclass={"install": EpicsInstall, "bdist": EpicsBDist})`.

**Comment by newville** [comment]
Nope, not changing setup.py to fit conda-forge.  Conda convert can use -f.

**Comment by mbargull** [comment]
Oh, that was unrelated to `conda-forge`, just meant as a general suggestion as it concern every PyEpics installation.
Once Travis CI's OSX queue is freed up a bit, you'll see that this recipe builds fine as is (assuming no OSX-related errors happen...).

**Comment by CJ-Wright** [comment]
Thank you @mbargull for all your help!

**Comment by mbargull** [comment]
Credits go to you and @newville! I merely took a look and made a comment, but thanks anyways 😄.

**Comment by CJ-Wright** [comment]
I've requested a rebuild on travis since the build was stopped due to an outage at travis

**Comment by newville** [comment]
@CJ-Wright  Thanks -- I hope it works this time!

**Comment by CJ-Wright** [comment]
Thank you everyone involved!

**Comment by newville** [comment]
@CJ-Wright Great! Thanks!

**Comment by CJ-Wright** [comment]
Of course, hopefully things get better as travis works on their Mac framework.

**Pull request merged**